maktabkhooneh-dl.py

```diff
@@ -133,22 +138,27 @@ def main():
                 logging.info('Maybe the page is a quize')
                 continue
 
-        # Download
-        logging.info("Downloading lecture file...")
-        download(make_absolute(download_link), intermediate_file)
+        if args.store_urls:
+            urls_file.write(make_absolute(download_link)+"\n")
 
-        # Convert/rename
-        if not args.mp3:
-            # No further processing needed -- just rename
-            os.rename(intermediate_file, final_file)
-        else:
-            # Convert
-            logging.info("Converting to mp3...")
-            subprocess.check_output(["ffmpeg", "-i", intermediate_file, final_file]) # TODO: Make non-blocking
-            os.remove(intermediate_file)
+        # Download
+        if not args.no_download:
+            logging.info("Downloading lecture file...")
+            download(make_absolute(download_link), intermediate_file)
+
+            # Convert/rename
+            if not args.mp3:
+                # No further processing needed -- just rename
+                os.rename(intermediate_file, final_file)
+            else:
+                # Convert
+                logging.info("Converting to mp3...")
+                subprocess.check_output(["ffmpeg", "-i", intermediate_file, final_file]) # TODO: Make non-blocking
+                os.remove(intermediate_file)
         
         logging.info("Finished lecture.")
 
+    urls_file.close()
```

Comment:
This line breaks when `args.store_urls` is False. Wrap it in an `if`.

Reply:
Thanks, that was my bad ;)